compiler/rustc_hir_analysis/src/variance/mod.rs

```diff
@@ -125,7 +118,8 @@ fn variance_of_opaque(tcx: TyCtxt<'_>, item_def_id: LocalDefId) -> &[ty::Varianc
                 // FIXME(-Zlower-impl-trait-in-trait-to-assoc-ty) check whether this is necessary
                 // at all for RPITITs.
                 ty::Alias(_, ty::AliasTy { def_id, substs, .. })
-                    if self.tcx.is_impl_trait_in_trait(*def_id) =>
+                    if self.tcx.is_impl_trait_in_trait(*def_id)
+                        && !self.tcx.lower_impl_trait_in_trait_to_assoc_ty() =>
```

Comment:
Removing this `visit_opaque` call with `-Zlower_impl_trait_in_trait_to_assoc_ty` allows us to remove the `variance_of_opaque` call above for RPITIT gats.

Reply:
The change on this line is otherwise identical to 9547302677f1dbd3694b4a477eeee7318d242f4c, except gated behind `lower_impl_trait_in_trait_to_assoc_ty` so it can land before the migration PR.